### PR TITLE
Remove `getPointerElementType` from DLLImportRelocation.cpp

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

File: gen/passes/DLLImportRelocation.cpp [hunk 1]
```diff
@@ -188,18 +188,21 @@ struct Impl {
...
       if (i <= 0xFFFFFFFFu) {
         address = b.CreateConstInBoundsGEP2_32(t, address, 0,
                                                static_cast<unsigned>(i));
       } else {
         address = b.CreateConstInBoundsGEP2_64(t, address, 0, i);
       }
+      if (StructType *st = dyn_cast<StructType>(t))
+        t = st->getElementType(i);
+      else if (PointerType *pt = dyn_cast<PointerType>(t))
+        assert(0 && "Have to give this some more thought");
```
Comment (thewilsonator) [inline gen/passes/DLLImportRelocation.cpp:202]: Hmm, I think I also need to account for `ArrayType` here too.

**Comment by thewilsonator** [comment]
Is this pass actually tested by the CI or am I just getting lucky here?

**Comment by kinke** [comment]
It is minimally tested by a few lit-tests at least. It should be absolutely safe to assume only IR structs and arrays of constant initializers are being GEP'd.

**Comment by thewilsonator** [comment]
Thanks. OK, I'll clean up the assert message and then merge this

**Pull request merged**